cai_causal_graph/graph_components.py

```diff
@@ -437,6 +491,23 @@ def details(self) -> str:
         """Return a detailed string description of the object."""
         return self.__repr__()
 
+    @classmethod
+    def from_dict(cls, edge_dict: dict) -> Edge:
+        """
+        Deserialize a dictionary to a `cai_causal_graph.graph_components.Edge` instance.
+
+        :param edge_dict: The dictionary representation of the `cai_causal_graph.graph_components.Edge` instance.
+        :return: The `cai_causal_graph.graph_components.Edge` instance.
+        """
+        source = Node.from_dict(edge_dict['source'])
+        destination = Node.from_dict(edge_dict['destination'])
```

Comment:
What if the nodes should be time series ones? Unit test this to see what happens when the original edge is between two time series nodes.

Reply:
As we call `to_dict` for the node, then I think we can tell by the `edge_dict['source']` dictionary if the node should be `Node` or `TimeSeriesNode`.